internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -430,6 +446,29 @@ func createWindowsContainerDocument(ctx context.Context, coi *createOptionsInter
 					DWordValue: dumpType,
 					Type_:      "DWord",
 				},
+				{
+					Key: &hcsschema.RegistryKey{
+						Hive: "System",
+						Name: "ControlSet001\\Control",
+					},
+					Name:        "WaitToKillServiceTimeout",
+					StringValue: strconv.Itoa(math.MaxInt32),
+					Type_:       "String",
+				},
+			},
+		}
+	} else {
+		v2Container.RegistryChanges = &hcsschema.RegistryChanges{
```

Comment:
I think it would be cleaner to change this to:

```golang
v2Container.RegistryChanges = &hcsschema.RegistryChanges{
    AddValues: []hcsschema.RegistryValue{
        {
            // Your key here.
        },
    },
}

if dumpPath != "" {
    [...]

    v2Container.RegistryChanges.AddValues = append(v2Container.RegistryChanges.AddValues, []hcsschema.RegistryValue{
        // Dump keys here
    }...)
}
```